cromwell-backend/src/main/scala/cromwell/backend/BackendFactory.scala

```diff
@@ -0,0 +1,26 @@
+package cromwell.backend
+
+import akka.actor.{ActorRef, ActorSystem}
+import cromwell.backend.model.TaskDescriptor
+
+/**
+  * Creates Backend instances based on backend name, actor system and task.
+  */
+trait BackendFactory {
+  /**
+    * Returns a backend instance based on the name and specific task.
+    * @param initClass Initialization class for the specific backend.
+    * @param task Specific task to be executed in the backend.
+    * @return A backend instance.
+    */
+  def getBackend(initClass: String, task: TaskDescriptor) : Backend
```

Comment:
This will be removed. I think we should be supporting only actor based backends as of now.